### PR TITLE
BAU - remove "page name" map from navigators

### PR DESCRIPTION
**Pull request by SteveSugden**

### CODE REVIEW

File: app/uk/gov/hmrc/nationalimportdutyadjustmentcentrefrontend/pages/Page.scala [hunk 1]
```diff
@@ -23,3 +23,8 @@ case object RequiredDocumentsPage extends Page
 case object UploadPage            extends Page
 case object CheckYourAnswersPage  extends Page
 case object ConfirmationPage      extends Page
+
+object Implicit {
```
Comment (crosslandwa) [inline app/uk/gov/hmrc/nationalimportdutyadjustmentcentrefrontend/pages/Page.scala:27]: CLARIFICATION: I'd appreciate a quick talk-through of how this works (it looks slightly voodoo-ish to my naive eyes). I assume type-checking is still provided in places that import `pages.Implicit._` then reference "pages"?

File: test/uk/gov/hmrc/nationalimportdutyadjustmentcentrefrontend/views/amendclaim/CheckYourAnswersViewSpec.scala [hunk 2]
```diff
@@ -90,7 +89,7 @@ class CheckYourAnswersViewSpec extends UnitViewSpec with TestData {
           s"${messages("site.change")} ${messages("amend.check_answers.information.uploadedDocuments.accessible")}"
         )
 
-        uploadRow must haveSummaryActionsHref(routes.CheckYourAnswersController.onChange(AmendPageNames.uploadSummary))
+        uploadRow must haveSummaryActionsHref(routes.CheckYourAnswersController.onChange(UploadPage))
```
Comment (crosslandwa) [inline test/uk/gov/hmrc/nationalimportdutyadjustmentcentrefrontend/views/amendclaim/CheckYourAnswersViewSpec.scala:92]: CLARIFICATION: How does one distinguish if `UploadPage` refers to the create claim upload page or the amend claim upload page? Does it matter?

**Review by crosslandwa — APPROVED** [review]

File: app/uk/gov/hmrc/nationalimportdutyadjustmentcentrefrontend/pages/Page.scala (hunk 1 above)
Comment (SteveSugden) [reply app/uk/gov/hmrc/nationalimportdutyadjustmentcentrefrontend/pages/Page.scala:27]: Yes.  We can have a chat later.   But places that require a `Page` will still enforce that a `Page` is provided.    This implicit type conversion is to allow a `Page` to be passed to a method that requires a `String` (without having to type `SomePage.toString`)

File: test/uk/gov/hmrc/nationalimportdutyadjustmentcentrefrontend/views/amendclaim/CheckYourAnswersViewSpec.scala (hunk 2 above)
Comment (SteveSugden) [reply test/uk/gov/hmrc/nationalimportdutyadjustmentcentrefrontend/views/amendclaim/CheckYourAnswersViewSpec.scala:92]: It doesn't matter.  As long as that `Page` only occurs once in the `pageOrder: Seq[P]` definition

**Review by crosslandwa — APPROVED** [review]

**Pull request merged**